.github/workflows/ci_workflow.yml

```diff
@@ -0,0 +1,52 @@
+name: CI Workflow
+on:
+  push:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+      - name: Check code formatting
+        run: make lint
+      - name: Run unit tests
+        run: |
+          touch .env
+          make test
+  deploy:
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/')
+    permissions:
+      id-token: write
+      contents: read
+    needs:
+      - test
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+      - name: Set Docker tag
+        id: docker_tag
+        env:
+          BRANCH: ${{ github.head_ref }}
+        run: |
+          if [[ "$BRANCH" == refs/tags/* ]]; then
+            echo "tag=$BRANCH" >> $GITHUB_OUTPUT
+          else
+            echo "tag=latest" >> $GITHUB_OUTPUT
+          fi
+      - name: Build the Docker image
+        env:
+          DOCKER_TAG: ${{ steps.docker_tag.outputs.tag }}
+        run: docker build . -t us-docker.pkg.dev/moz-fx-data-artifacts-prod/ensemble-transposer/ensemble-transposer:$DOCKER_TAG
```

Comment:
https://github.com/mozilla/telemetry-airflow/blob/7519f70d31ee7a206e1594fa4f57c202ace1dfd2/dags/firefox_public_data_report.py#L155 will need to be updated too

Reply:
https://github.com/mozilla/telemetry-airflow/pull/2281